File: ios/TicTacToe/Game/States/GameOverState.swift
import GameplayKit

/// Terminal state for one match — see `docs/GAME_DESIGN.md`'s "Turn flow" section: a new game is
/// a new `TicTacToeMatch`, never a transition back out of this state.
final class GameOverState: GKState {
    override func isValidNextState(_ stateClass: AnyClass) -> Bool {
        false
    }
}

File: android/app/src/main/kotlin/jp/co/bitz/tictactoe/game/states/GameOverState.kt
package jp.co.bitz.tictactoe.game.states

import jp.co.bitz.gameplaykit.GKState
import kotlin.reflect.KClass

/**
 * Terminal state for one match — see docs/GAME_DESIGN.md's "Turn flow" section: a new game is a
 * new [jp.co.bitz.tictactoe.game.TicTacToeMatch], never a transition back out of this state.
 */
public class GameOverState : GKState() {
    override fun isValidNextState(stateClass: KClass<out GKState>): Boolean = false
}
